carbonmark/components/pages/Project/Purchase/PurchaseInputs.tsx

```diff
@@ -0,0 +1,186 @@
+import { LO } from "lib/luckyOrange";
+import { CarbonmarkPaymentMethod, Listing } from "lib/types/carbonmark";
+import Image from "next/legacy/image";
+import { useRouter } from "next/router";
+import { FC } from "react";
+import { SubmitHandler, useFormContext } from "react-hook-form";
+import * as styles from "./styles";
+import { FormValues } from "./types";
+
+type Props = {
+  onSubmit: (values: FormValues) => void;
+  listing: Listing;
+  values: null | FormValues;
+  balance: string | null;
+};
+
+export const PurchaseInputs: FC<Props> = (props) => {
+  const { locale } = useRouter();
+
+  const { register, handleSubmit, formState, control, clearErrors } =
+    useFormContext<FormValues>();
+
+  const onSubmit: SubmitHandler<FormValues> = (values: FormValues) => {
+    LO.track("Purchase: Continue Clicked");
+    props.onSubmit(values);
+  };
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <div className={styles.inputsContainer}>
+        <InputField
+          id="listingId"
+          inputProps={{
+            type: "hidden",
+            ...register("listingId"),
+          }}
+          label={"listing ID"}
+          hideLabel
+        />
+
+        <div className={styles.amountLabel}>
+          <Text>{t`How many tonnes of carbon do you want to buy?`}</Text>
+          <Text t="body3">
+            <Trans>Available:</Trans> {formatUnits(props.listing.leftToSell)}
+          </Text>
+        </div>
+
+        <InputField
+          id="amount"
+          inputProps={{
+            placeholder: t({
+              id: "purchase.input.amount.placeholder",
+              message: "Tonnes",
+            }),
+            type: "number",
+            min: 1,
+            max: Number(formatUnits(props.listing.leftToSell)),
+            ...register("amount", {
+              onChange: () => clearErrors("price"),
+              required: {
+                value: true,
+                message: t({
+                  id: "purchase.input.amount.required",
+                  message: "Amount is required",
+                }),
+              },
+              min: {
+                value: 1,
+                message: t({
+                  id: "purchase.input.amount.minimum",
+                  message: "The minimum amount to buy is 1 Tonne",
+                }),
+              },
+              max: {
+                value: Number(formatUnits(props.listing.leftToSell)),
+                message: t({
+                  id: "purchase.input.amount.maxAmount",
+                  message: "You exceeded the available amount of tonnes",
+                }),
+              },
+            }),
+          }}
+          label={t`How many tonnes of carbon do you want to buy?`}
+          errorMessage={formState.errors.amount?.message}
+          hideLabel
+        />
+
+        <div className={styles.paymentLabel}>
+          <Text>{t`Pay with:`}</Text>
+          <Text t="body3">
+            {t`Balance:`}{" "}
```

Comment:
Interpolate the value into the string because other languages might format this differently
```suggestion
            {t`Balance: ${balance}`}
```